[CI] RandomizedRollingUpgradeIT testIndexingSyntheticSource failing

**Build Scans:**
- [elasticsearch-pull-request #114269 / 9.3.0_part_2_bwc-snapshots](https://gradle-enterprise.elastic.co/s/4ct26t5ufcweo)
- [elasticsearch-pull-request #114269 / 9.1.11_part_2_bwc-snapshots](https://gradle-enterprise.elastic.co/s/mabgzaehruphq)
- [elasticsearch-pull-request #114269 / 9.2.5_part_2_bwc-snapshots](https://gradle-enterprise.elastic.co/s/zugrrliaqmbw4)
- [elasticsearch-pull-request #114269 / 8.19.11_part_2_bwc-snapshots](https://gradle-enterprise.elastic.co/s/omdqh4xdzkqii)
- [elasticsearch-intake #33248 / 9.1.11_bwc-snapshots](https://gradle-enterprise.elastic.co/s/iomkoprit4giw)
- [elasticsearch-pull-request #114257 / 8.19.11_part_2_bwc-snapshots](https://gradle-enterprise.elastic.co/s/thhmnwib6gcdi)
- [elasticsearch-pull-request #114257 / 9.1.11_part_2_bwc-snapshots](https://gradle-enterprise.elastic.co/s/ihcji7fcm3aeq)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:logsdb:qa:rolling-upgrade:v9.1.11#bwcTest" -Dtests.class="org.elasticsearch.xpack.logsdb.RandomizedRollingUpgradeIT" -Dtests.method="testIndexingStandardSource" -Dtests.seed=D60480A87D19BA8 -Dtests.bwc=true -Dtests.locale=en-RO -Dtests.timezone=Etc/GMT-7 -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.logsdb.RandomizedRollingUpgradeIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testIndexingSyntheticSource'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: timed out waiting for green state for index [test-index-standard-*] cluster state [{
  "cluster_name" : "test-cluster",
  "cluster_uuid" : "k_kLr1UiR-SxaPtM3WhIZw",
  "version" : 55,
  "state_uuid" : "jb6xgtP_Ty-sa6Yw0_5zyQ",
  "master_node" : "gK4-hbKQT0uzUWweQfXIjA",
  "blocks" : { },
  "nodes" : {
    "xiRbZmorSmiAVWZQ2Y2xWQ" : {
      "name" : "test-cluster-1",
      "ephemeral_id" : "cylzvANvTr6ANU9ncfQscg",
      "transport_address" : "127.0.0.1:44417",
      "external_id" : "test-cluster-1",
      "attributes" : {
        "transform.config_version" : "10.0.0",
        "testattr" : "test",
        "ml.allocated_processors" : "2",
        "ml.max_jvm_size" : "805306368",
        "ml.config_version" : "12.0.0",
        "ml.allocated_processors_double" : "2.0",
        "ml.machine_memory" : "101238525952",
        "xpack.installed" : "true"
      },
      "roles" : [
        "data",
        "data_cold",
        "data_content",
        "data_frozen",
       
[truncated]
```

**Issue Reasons:**
- [main] 7 failures in test testIndexingSyntheticSource (0.7% fail rate in 1000 executions)
- [main] 2 failures in step 9.1.11_part_2_bwc-snapshots (1.7% fail rate in 120 executions)
- [main] 2 failures in step 8.19.11_part_2_bwc-snapshots (1.7% fail rate in 119 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (1.3% fail rate in 151 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -394,9 +394,6 @@ tests:
 - class: org.elasticsearch.persistent.ClusterPersistentTasksCustomMetadataTests
   method: testMinVersionSerialization
   issue: https://github.com/elastic/elasticsearch/issues/139741
-- class: org.elasticsearch.xpack.logsdb.RandomizedRollingUpgradeIT
-  method: testIndexingSyntheticSource
-  issue: https://github.com/elastic/elasticsearch/issues/139482
 - class: org.elasticsearch.test.rest.yaml.CssSearchYamlTestSuiteIT
   method: test {p0=search.retrievers/result-diversification/10_mmr_result_diversification_retriever/Test MMR result diversification multiple indexes}
   issue: https://github.com/elastic/elasticsearch/issues/139826

--- a/x-pack/plugin/logsdb/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/RandomizedRollingUpgradeIT.java
+++ b/x-pack/plugin/logsdb/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/RandomizedRollingUpgradeIT.java
@@ -132,6 +132,7 @@ public class RandomizedRollingUpgradeIT extends AbstractLogsdbRollingUpgradeTest
 
         int numNodes = Integer.parseInt(System.getProperty("tests.num_nodes", "3"));
         for (int i = 0; i < numNodes; i++) {
+            flush(indexNameBase + "*", true);
             upgradeNode(i);
             ensureGreen(indexNameBase + "*");
             for (int j = 0; j < NUM_INDICES; j++) {

